software/base/src/main/java/org/apache/brooklyn/entity/software/base/VanillaWindowsProcessWinRmDriver.java

```diff
@@ -37,31 +42,36 @@ public void start() {
 
         super.start();
     }
-    
+
     @Override
-    public void preInstall() {
-        super.preInstall();
-        executeCommand(VanillaWindowsProcess.PRE_INSTALL_COMMAND, VanillaWindowsProcess.PRE_INSTALL_POWERSHELL_COMMAND, true);
-        if (entity.getConfig(VanillaWindowsProcess.PRE_INSTALL_REBOOT_REQUIRED)) {
-            rebootAndWait();
+    public void runPreInstallCommand() {
+        if(Strings.isNonBlank(getEntity().getConfig(VanillaWindowsProcess.PRE_INSTALL_COMMAND)) || Strings.isNonBlank(getEntity().getConfig(VanillaWindowsProcess.PRE_INSTALL_POWERSHELL_COMMAND))) {
+            executeCommand(VanillaWindowsProcess.PRE_INSTALL_COMMAND, VanillaWindowsProcess.PRE_INSTALL_POWERSHELL_COMMAND, true);
+            if (entity.getConfig(VanillaWindowsProcess.PRE_INSTALL_REBOOT_REQUIRED)) {
```

Comment:
If one had configured `PRE_INSTALL_REBOOT_REQUIRED` without any pre install command, we should still execute it. Let's move this check outside of the outer if block.

Also, I'd use `if (Boolean.TRUE.equals(entity.getConfig(VanillaWindowsProcess.PRE_INSTALL_REBOOT_REQUIRED))) {`. Otherwise there is a risk of a NullPointerException if the config parameter's value is null.